### PR TITLE
Allow multiple API keys management with profiles

### PR DESCRIPTION
**Pull request by bonidier**
Hi @denouche , Thanks for your script, very useful !

Here, I've resolved some limitations :

-  Your script can use only one pair of Application+Consumer Keys, but we may need to use it for many OVH accounts or uses.
-  When you init keys, it force all grants to the API : It may be dangerous if using your client on a server with automated tasks.

Regards

### CODE REVIEW

**Comment by bonidier** [comment]
Hi @denouche , Thanks for your script, very useful !

Here, I've resolved some limitations : 
- Your script can use only one pair of Application+Consumer Keys, but we may need to use it for many OVH accounts or uses.
- When you init keys, it force all grants to the API : It may be dangerous if using your client on a server with automated tasks.


Regards

File: ovh-api-bash-client.sh [hunk 1]
```diff
@@ -201,28 +229,70 @@ getJSONFieldString()
...
+    if [ ! -d "${PROFILES_PATH}/${profile}" ]
+    then
+     echo "${PROFILES_PATH}/${profile} should exists"
+     exit 1
+    fi
+    # override default configuration location
+    CURRENT_PATH="$( cd "${PROFILES_PATH}/${profile}" && pwd )"
+    HELP_CMD="${HELP_CMD} --profile ${profile}"
+  else
+    echo "profile not set, choose in : "
```
Comment (denouche) [inline ovh-api-bash-client.sh]: In this case maybe we could use a `default` profile, to avoid these modifications to be breaking for existing users?

**Comment by denouche** [comment]
Wow, really nice suggestions!
Thank you for your PR.

I only added 1 comment. Let me know what you think about my suggestion.

File: ovh-api-bash-client.sh (hunk 1 above)
Comment (bonidier) [reply ovh-api-bash-client.sh]: @denouche this doesn't break users : if --profile is not set, the keys files goes to script location. 
(the CURRENT_PATH in only overrided in the initProfile function)

I can :
- simulate an explicit profile _default_  : if setted to that value, or without --profile xxx
- moving the default location (CURRENT_PATH value) to the initProfile function.

It should be more logical.

File: ovh-api-bash-client.sh [hunk 2]
```diff
@@ -18,8 +18,13 @@ API_URLS[EU]="https://api.ovh.com/1.0"
 declare -A API_CREATE_APP_URLS
 API_CREATE_APP_URLS[CA]="https://ca.api.ovh.com/createApp/"
 API_CREATE_APP_URLS[EU]="https://api.ovh.com/createApp/"
-CURRENT_PATH="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
+BASE_PATH="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+CURRENT_PATH="${BASE_PATH}"
+#PROFILES_PATH
```
Comment (bonidier) [inline ovh-api-bash-client.sh]: I should remove that

**Comment by bonidier** [comment]
Hi @denouche, I've made changes like expected : the virtual profile "default" is now mapped to script directory (like if you don't set --profile argument)

Regards

**Comment by denouche** [comment]
Nice.

My last 2 comments:
- if you execute: `$ rm -rf ./profile && ./ovh-api-bash-client.sh --profile test`, the folders listed are the folders in the base path instead of just `default`, the result is:
```
Available profiles : 
- default
- libs
- profile
```

- then if I call `./ovh-api-bash-client.sh --profile test --initApp` it also just list the available profiles and quit. Around line 254, maybe we could create the profile folder if the `INIT_KEY_ACTION` is set? 
That's okay to list the profile and quit in "normal" usage, but when initializing I think it could be nice to create the folder for the user and continue the initialization.

**Comment by bonidier** [comment]
Hi @denouche, thanks for your feedback !

I've solved your requests, It's working fine now.

Good idea for the profile creation when initializing, it avoid the user to run the mkdir before.

Regards

**Comment by denouche** [comment]
Thank you for your contribution!

:+1:

**Pull request merged**